sdk/aqueduct/decorator.py

```diff
@@ -250,16 +281,27 @@ def wrapped(*input_artifacts: BaseArtifact) -> BaseArtifact:
                 OperatorSpec(function=function_spec),
                 *input_artifacts,
                 op_name=name,
+                description=description,
+                execution_mode=execution_mode,
             )
 
             return new_function_artifact
 
+        def wrapped(*input_artifacts: BaseArtifact) -> BaseArtifact:
+            return _wrapped_util(*input_artifacts, execution_mode=ExecutionMode.EAGER)
+
         # Enable the .local(*args) attribute, which calls the original function with the raw inputs.
-        def local_func(*inputs: Any) -> DataFrame:
+        def local_func(*inputs: Any) -> Any:
```

Comment:
just for my own learning, what is `local_func` for? Also I slightly prefer `local` or `local_op` to make concepts consistent

Reply:
Another question is - does these expose `.local` and `.lazy` for all wrapped artifacts or we only enable these method under specific modes? If it's the former case I feel it's hacky to use a `setattr` call. It seems that we could just expose `.local` and `.lazy` from `Artifact` objects?

Reply:
`local_func` is used when the user just wants to use a decorated function as if it's a normal function. Maybe we can change the name to `local_mode`?

`.local` and `.lazy` aren't attached to an artifact; they are attached to a decorated function. So say we defined a function as:
```
@op 
def func(args):
  return "hello"
```

then `bar = func(foo)` executes in eager mode, `bar = func.lazy(foo)` executes in lazy mode.

Reply:
Yeah both `.local` `.lazy` or `.local_mode` `.lazy_mode` makes sense, feels more consistent. Thanks for explaining how it's applied to operators, in this case I would try to make the `@op` wrapper returning an object with `__call__`, `.lazy()` and `.local` implemented so that it's more clear on what attributes are available. I don't think we need to do it in this PR but happy to provide more details as follow-up